Linux gitian build fails in SHA256AutoDetect: inconsistent operand constraints in an ‘asm’

Full log: https://bitcoin.jonasschnelli.ch/builds/667/build_linux.log
Excerpt:
```
crypto/sha256.cpp: In function ‘std::string SHA256AutoDetect()’:
crypto/sha256.cpp:537:83: error: inconsistent operand constraints in an ‘asm’
   __asm__ ("cpuid" : "=a"(a), "=b"(b), "=c"(c), "=d"(d) : "0"(leaf), "2"(subleaf));
                                                                                   ^
crypto/sha256.cpp:537:83: error: inconsistent operand constraints in an ‘asm’
   __asm__ ("cpuid" : "=a"(a), "=b"(b), "=c"(c), "=d"(d) : "0"(leaf), "2"(subleaf));
                                                                                   ^
make[2]: *** [crypto/crypto_libbitcoin_crypto_base_a-sha256.o] Error 1
make[2]: *** Waiting for unfinished jobs....
  CXXLD    libunivalue.la
make[3]: Leaving directory `/home/ubuntu/build/bitcoin/distsrc-i686-pc-linux-gnu/src/univalue'
make[2]: Leaving directory `/home/ubuntu/build/bitcoin/distsrc-i686-pc-linux-gnu/src'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/home/ubuntu/build/bitcoin/distsrc-i686-pc-linux-gnu/src'
make: *** [all-recursive] Error 1

--- a/src/crypto/sha256.cpp
+++ b/src/crypto/sha256.cpp
@@ -9,7 +9,7 @@
 #include <string.h>
 #include <atomic>
 
-#if defined(__x86_64__) || defined(__amd64__)
+#if defined(__x86_64__) || defined(__amd64__) || defined(__i386__)
 #if defined(USE_ASM)
 #include <cpuid.h>
 namespace sha256_sse4
@@ -534,7 +534,11 @@ bool SelfTest() {
 // We can't use cpuid.h's __get_cpuid as it does not support subleafs.
 void inline cpuid(uint32_t leaf, uint32_t subleaf, uint32_t& a, uint32_t& b, uint32_t& c, uint32_t& d)
 {
+#ifdef __GNUC__
+    __cpuid_count(leaf, subleaf, a, b, c, d);
+#else
   __asm__ ("cpuid" : "=a"(a), "=b"(b), "=c"(c), "=d"(d) : "0"(leaf), "2"(subleaf));
+#endif
 }
 
 /** Check whether the OS has enabled AVX registers. */

